### PR TITLE
Switch to eclipse temurin, build via jlink, switch to bullseye

### PR DESCRIPTION
**Pull request by timja**
See https://github.com/jenkinsci/docker/pull/1185

### CODE REVIEW

File: 11/archlinux/Dockerfile [hunk 1]
```diff
@@ -20,7 +20,22 @@
 #  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 #  THE SOFTWARE.
 
-FROM archlinux:latest AS archlinux
+FROM archlinux:latest AS jre-build
```
Comment (MarkEWaite) [inline 11/archlinux/Dockerfile:23]: This is intentionally not coming from Eclipse Temurin?
```suggestion
FROM archlinux:latest AS jre-build
```

**Review by MarkEWaite — COMMENTED** [review]
One question on the archlinux change

File: 11/archlinux/Dockerfile (hunk 1 above)
Comment (timja) [reply 11/archlinux/Dockerfile:23]: I didn’t bother no, I question whether this image should be here at all.

I wasn’t sure if there was implications from getting it from a different OS type like Debian whether libraries would be different…

**Review by MarkEWaite — APPROVED** [review]

**Review by dduportal — APPROVED** [review]
🚀 Good work, sounds really good! (the glibc part on Alpine is tricky but needed until eclipse-temurin provides their own Alpine - https://github.com/adoptium/temurin-build/projects/1?fullscreen=true&card_filter_query=label%3Aalpine-linux)

**Comment by MarkEWaite** [comment]
> 🚀 Good work, sounds really good! (the glibc part on Alpine is tricky but needed until eclipse-temurin provides their own Alpine - https://github.com/adoptium/temurin-build/projects/1?fullscreen=true&card_filter_query=label%3Aalpine-linux)

Eclipse has provided their own alpine as an unofficial image.  We used it previously.  However, I like the transition to Eclipse Temurin even more than using unofficial Adoptium images.

**Pull request merged**